demisto_sdk/commands/download/downloader.py

```diff
@@ -237,189 +227,214 @@ def verify_flags(self) -> bool:
 
-    def replace_uuids(
-        self, string_to_write: str, uuid_dict: dict, file_name: str
-    ) -> str:
+    def should_download_playbook(self, playbook_data: StringIO) -> str | None:
         """
-        Replace all occurrences of UUIDs in a string with their corresponding values from a dictionary.
+        Check if the content item should be downloaded separately using a file-specific API call.
 
-        Parameters:
-        - string_to_write (str): The string to search for UUIDs in.
-        - uuid_dict (dict): A dictionary mapping UUIDs to content item IDs.
+        Note:
+            This method is used to determine whether a playbook should be downloaded separately from the content bundle.
+            This is needed since properties like 'scriptName' are emitted from playbooks on the content bundle.
+
+        Args:
+            playbook_data (StringIO): The content of the file to check.
 
         Returns:
-        - str: The modified string with all UUIDs replaced.
+            str | None: Playbook ID if the playbook should be downloaded separately, None otherwise.
         """
-        uuids = re.findall(UUID_REGEX, string_to_write)
+        if self.list_files:
+            return None
 
-        for uuid in set(uuids).intersection(uuid_dict):
-            logger.debug(
-                f"Replacing UUID: {uuid} with the following:\
- {uuid_dict[uuid]} in {file_name}"
-            )
-            string_to_write = string_to_write.replace(uuid, uuid_dict[uuid])
-        return string_to_write
+        playbook_data = yaml.load(playbook_data)
+        playbook_id = playbook_data.get("id")
+        playbook_name = playbook_data.get("name")
 
-    def should_download_playbook(self, file_name: str) -> bool:
-        #  if the content item is playbook and list-file flag is true, we should download the
-        #  file via direct REST API because there are props like scriptName, that playbook from custom
-        #  content bundle don't contain
-        return bool(not self.list_files and re.search(PLAYBOOK_REGEX, file_name))
+        # If missing a needed field, or playbook is not supposed to be downloaded
+        if not all((playbook_id, playbook_name)) or \
+                (playbook_name not in self.input_files) and (playbook_name not in self.all_custom_content):
+            return None
 
-    def write_custom_content(
-        self, content_items_file_names: List[Tuple[str, str]], scripts_id_to_name: dict
-    ):
-        for content_item_as_string, file_name in content_items_file_names:
-            if self.should_download_playbook(file_name):
-                content_item_as_string = self.download_playbook_yaml(
-                    content_item_as_string
-                )
-            content_item_as_string = self.replace_uuids(
-                content_item_as_string, scripts_id_to_name, file_name
-            )
-            file_name = self.update_file_prefix(file_name.strip("/"))
```

Comment:
Self note: Remember to use this method (replaces `automation-` prefix with `script-`) when saving files from memory to disk.